src/components/brands/TrainingSelectFixed.jsx

```
@@ -0,0 +1,373 @@
+  const searchTimeoutRef = useRef(null);
+  const requestIdRef = useRef(0);
+  const fetchAbortControllerRef = useRef(null);
+
+  // Load initial trainings and selected training data
+  useEffect(() => {
+    if (brandId) {
+      // Reset request tracking when brandId changes
+      requestIdRef.current = 0;
+      loadTrainings('');
+    }
+  }, [brandId, loadTrainings]);
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)